### ISSUE TITLE
module 'prophet.hdays' has no attribute 'UK'

### ISSUE THREAD

**Issue by bowie-tey** [facebook/prophet#2120]
Here is my code below, I'm not too sure why this issue is occurring, I have the latest holidays package installed on python version 0.12. I'm assuming the string I've entered 'UK' is wrong for United Kingdom ? since changing it to FR and US the error stops.
```
m = Prophet(holidays= holidays, weekly_seasonality=True)
m.add_country_holidays(country_name='UK')
m.fit(prophet_df)`
```

Here is the error output

>  TypeError                                 Traceback (most recent call last)
> /var/folders/t0/nm4jf38s55d57xjkvlp809vc0000gn/T/ipykernel_1698/1020037414.py in <module>
>       1 m = Prophet(holidays= holidays, weekly_seasonality=True)
>       2 m.add_country_holidays(country_name='UK')
> ----> 3 m.fit(prophet_df)
> 
> ~/miniforge3/envs/ds/lib/python3.8/site-packages/prophet/forecaster.py in fit(self, df, **kwargs)
>    1118         self.set_auto_seasonalities()
>    1119         seasonal_features, prior_scales, component_cols, modes = (
> -> 1120             self.make_all_seasonality_features(history))
>    1121         self.train_component_cols = component_cols
>    1122         self.component_modes = modes
> 
> ~/miniforge3/envs/ds/lib/python3.8/site-packages/prophet/forecaster.py in make_all_seasonality_features(self, df)
>     782 
>     783         # Holiday features
> --> 784         holidays = self.construct_holiday_dataframe(df['ds'])
>     785         if len(holidays) > 0:
>     786             features, holiday_priors, holiday_names = (
> 
> ~/miniforge3/envs/ds/lib/python3.8/site-packages/prophet/forecaster.py in construct_holiday_dataframe(self, dates)
>     488         if self.country_holidays is not None:
>     489             year_list = list({x.year for x in dates})
> --> 490             country_holidays_df = make_holidays_df(
>     491                 year_list=year_list, country=self.country_holidays
>     492             )
> 
> ~/miniforge3/envs/ds/lib/python3.8/site-packages/prophet/make_holidays.py in make_holidays_df(year_list, country, province, state)
>      58     except AttributeError:
>      59         try:
> ---> 60             holidays = getattr(hdays_part1, country)(prov=province, state=state, years=year_list, expand=False)
>      61         except AttributeError as e:
>      62             raise AttributeError(
> 
> ~/miniforge3/envs/ds/lib/python3.8/site-packages/holidays/countries/united_kingdom.py in __init__(self, **kwargs)
>      34         if "state" not in kwargs:
>      35             kwargs["state"] = "UK"
> ---> 36         HolidayBase.__init__(self, **kwargs)
>      37 
>      38     def _populate(self, year):
> 
> ~/miniforge3/envs/ds/lib/python3.8/site-packages/holidays/holiday_base.py in __init__(self, years, expand, observed, prov, state)
>     180         self.state = state
>     181         for year in self.years.copy():
> --> 182             self._populate(year)
>     183 
>     184     def __setattr__(self, key: str, value: Any) -> None:
> 
> ~/miniforge3/envs/ds/lib/python3.8/site-packages/holidays/countries/united_kingdom.py in _populate(self, year)
>     107 
>     108         # Overwrite to modify country specific holidays
> --> 109         self._country_specific(year)
>     110 
>     111     def _country_specific(self, year):
> 
> ~/miniforge3/envs/ds/lib/python3.8/site-packages/holidays/countries/united_kingdom.py in _country_specific(self, year)
>     158 
>     159         # Late Summer bank holiday (last Monday in August)
> --> 160         if self.state not in ("Scotland") and year >= 1971:
>     161             name = "Late Summer Bank Holiday"
>     162             if self.state == "UK":
> 
> TypeError: 'in <string>' requires string as left operand, not NoneType

**Comment by camilleri-keith** [comment]
I've had this issue this morning as well.  The problem seems to be that Prophet is not compatible with the latest version of the 'holidays' library released January 4th ([https://pypi.org/project/holidays/#history](url)) - quickest (but should be temporary I guess) solution I found was to downgrade the holidays library to v0.11.3.1.

From the little debugging I did, it seems that the latest version of the holidays library also expects a state for the UK (ex. Scotland), which currently prophet passes as `None`, thus breaking this bit of code:

> ~/miniforge3/envs/ds/lib/python3.8/site-packages/holidays/countries/united_kingdom.py in _country_specific(self, year)
158
159 # Late Summer bank holiday (last Monday in August)
--> 160 if self.state not in ("Scotland") and year >= 1971:
161 name = "Late Summer Bank Holiday"
162 if self.state == "UK":

**Comment by bowie-tey** [comment]
Thanks that worked ! out of curiosity, how did you figure that out @camilleri-keith ?

**Comment by camilleri-keith** [comment]
Starting from the top of the error traceback, I realised that the issue could be either in my code, in the prophet library, or in the holidays library.  I encountered this error when I created a new virtual environment and installed all the required libraries, and so I was quite certain that my code wasn't the issue as I also had a fully functional version of it on an older environment - and so I blamed prophet/holidays.  I tried finding out if it is possible to pass a state alongisde the country to Prophet but couldn't, so I compared versions for prophet and holidays with a colleague and found different version numbers for holidays, so I downgraded the holidays library.
